src/mpid/ch4/netmod/ofi/ofi_spawn.c

```diff
@@ -19,6 +19,24 @@
 #define DYNPROC_RECEIVER 0
 #define DYNPROC_SENDER 1
 
+/* The normal progress is protected with vci_lock, the progress here is not and it needs to.
+ * We customized the OFI PROGRESS macros here to add the vci_locks as a work-around.
+ * FIXME: what is wrong is the yielding of MPIR_THREAD_GLOBAL_ALLFUNC_MUTEX. Lower-layer
+ *        does not know and should not know what outer-layer lock has been taken.
+ *        If structured properly, there shouldn't be yield.
+ */
+#define _fixme_MPIDI_OFI_PROGRESS()                                      \
+    do {                                                          \
+        MPID_THREAD_CS_ENTER(VCI, MPIDI_global.vci_lock);         \
+        mpi_errno = MPIDI_OFI_progress(0, 0);                     \
+        MPID_THREAD_CS_EXIT(VCI, MPIDI_global.vci_lock);         \
+        MPIR_ERR_CHECK(mpi_errno);                                \
+        MPID_THREAD_CS_YIELD(GLOBAL, MPIR_THREAD_GLOBAL_ALLFUNC_MUTEX); \
```

Comment:
@hzhou Was this yield needed as a result of using the `VCI` lock?

Reply:
Right now, yes, because the whole `MPI_Comm_spawn_multiple` is still protected by the `ALLFUNC_MUTEX` and we need yield to ensure it does not block other thread's progress. Once we remove that lock -- by adding appropriate granular locks -- then the yield will be unnecessary.

Reply:
Okay.

Reply:
Is your plan to merge it now and address the `fixme` part in future?

Reply:
If it alright, I would like to merge now and "fixme" later. It is easier to revisit this "yield" when the rest of per-vci patches gets merged in, especially the new per-vci progress code.